### PR TITLE
docs(documentation): README + getting-started install overhaul, add make link-plugins

### LINKED ISSUE #487: bug(testing): Surge XT plugin path is hardcoded — cross-platform discovery needed
## Context

Raised in [#481 review](https://github.com/tinaudio/synth-setter/pull/481#discussion_r3027967607). The getting-started docs say to "note the installation path" after installing Surge XT, but the test suite hardcodes platform-specific paths:

- `tests/data/vst/test_preset_params.py` → `/usr/lib/vst3/Surge XT.vst3` (Linux .deb path)
- `tests/docker/test_smoke.py` → `/usr/lib/vst3/Surge XT.vst3`
- `pipeline/schemas/config.py` default → `plugins/Surge XT.vst3` (relative symlink convention)
- `src/data/vst/generate_vst_dataset.py` CLI default → `plugins/Surge XT.vst3`

On macOS and Windows (or Linux with a non-default install prefix), `pytest -m requires_vst` silently skips all VST tests because the plugin isn't found at the hardcoded path.

## Canonical platform paths

| Platform | Path |
|---|---|
| Linux (.deb) | `/usr/lib/vst3/Surge XT.vst3` |
| Linux (manual) | `~/.vst3/Surge XT.vst3` or `/usr/local/lib/vst3/Surge XT.vst3` |
| macOS (system) | `/Library/Audio/Plug-Ins/VST3/Surge XT.vst3` |
| macOS (user) | `~/Library/Audio/Plug-Ins/VST3/Surge XT.vst3` |
| Windows | `C:\Program Files\Common Files\VST3\Surge XT.vst3` |
| Docker image | `/usr/lib/vst3/Surge XT.vst3` (from .deb in Dockerfile) |

### Option A: Auto-discover canonical paths + symlink to `plugins/`

A setup script (or make target) searches the canonical platform paths above, finds the installed Surge XT, and creates a symlink at `plugins/Surge XT.vst3`. All existing code already references this relative path for pipeline use.

**Pros:**
- Zero config for users who installed Surge XT via the standard installer
- The `plugins/` symlink convention already exists in pipeline config defaults
- One-time setup, works across test suite and pipeline

**Cons:**
- Symlinks don't work well on Windows without developer mode
- Adds a setup step users could forget (silent failure → tests skip)
- Fragile if Surge XT is installed in a non-canonical location (e.g., custom prefix)

### Option B: Environment variable (`SURGE_XT_PATH`)

Introduce an env var (e.g., `SURGE_XT_PATH`) that overrides the plugin path. Tests and pipeline code read this var, falling back to the current default.

**Pros:**
- Explicit, no magic — user controls exactly what path is used
- Works on all platforms including CI with custom install locations
- Easy to document, easy to set in `.env` or CI matrix

**Cons:**
- Every contributor must set it (or tests silently skip)
- Doesn't solve the "I installed it but forgot to set the var" problem
- Another env var to document and maintain

### Option C: Runtime search (no symlink)

At test collection time (conftest fixture) and in the pipeline/CLI, search the canonical paths list in order and use the first match. No symlink required.

**Pros:**
- Zero-config on all platforms — just install Surge XT and it works
- No symlinks, no env vars, no setup script
- Graceful: if not found, tests skip with a clear message listing searched paths

**Cons:**
- Implicit — harder to debug when the wrong copy is picked up (e.g., old version in system path vs new in user path)
- Platform-specific path list needs maintenance as Surge XT or OS conventions change
- Slower test collection (filesystem probes), though negligible in practice

### Option D: Combination — env var with auto-discover fallback

`SURGE_XT_PATH` takes precedence if set. Otherwise, search canonical paths. If nothing found, skip with a diagnostic message listing all searched locations.

**Pros:**
- Best of B and C: zero-config for standard installs, explicit override for edge cases
- CI can pin the path via env var; local devs get auto-discovery
- Clear error messages when nothing is found

**Cons:**
- Most complex to implement (though still modest)
- Two code paths to test and document

## Recommendation

Option D (env var + auto-discover fallback) gives the best developer experience. The implementation is a single `discover_plugin()` utility used by both the test conftest and the pipeline config defaults.

## Affected files

- `tests/data/vst/test_preset_params.py` — hardcoded `PLUGIN_PATH`
- `tests/docker/test_smoke.py` — hardcoded `PLUGIN_PATH`
- `pipeline/schemas/config.py` — default `plugin_path`
- `src/data/vst/generate_vst_dataset.py` — CLI default `--plugin_path`
- `tests/pipeline/conftest.py` — fixture `plugin_path`
- `docs/getting-started.md` — installation instructions

Refs #481

### LINKED ISSUE #601: docs(documentation): README + getting-started install overhaul — make install canonical, add make link-plugins
## Summary

Rewrite README install section and deduplicate against `docs/getting-started.md` §2. Establish the canonical install flow, frame pip/conda as alternatives, add a `make link-plugins` Make target for VST3 symlink setup, move Codespaces/Docker to the bottom of the README, and add devcontainer-as-root instructions.

Supports the MVP public-release milestone — a fresh-machine install must Just Work from the README.

## Canonical install flow

The README (and getting-started §2) should converge on a single canonical flow:

```bash
# 1. Install uv
curl -LsSf https://astral.sh/uv/install.sh | sh

# 2. Create and activate a Python 3.10 venv
uv venv --python 3.10 .venv
source .venv/bin/activate

# 3. Install the project and all dev tooling
make install

# 4. Symlink the Surge XT VST3 into plugins/
make link-plugins
```

pip and conda remain supported as interchangeable alternatives for users who manage their own Python environment, but `make install` is the recommended path — it already pulls in both `requirements-torch.txt` and `requirements-app.txt` (including pre-commit), so there's no need to document dev-tool install separately.

### README rewrite

- Promote `make install` as the canonical install command.
- Frame uv / pip / conda as interchangeable front-ends for package resolution; spell out the explicit uv path once as the recommended walkthrough.
- Deduplicate against `docs/getting-started.md` §2 — pick one canonical location (likely getting-started) and have README link to it with a short summary, rather than repeating the full walkthrough.
- Move Codespaces and Docker sections to the bottom of the README (they're specialized paths, not the primary story).
- Add a devcontainer-as-root section — short note on running the devcontainer as root when the default non-root setup doesn't fit.
- Add an "Export your env vars" section with the `set -a && source .env; set +a` pattern (the env var docs overhaul sibling will go into more detail; this is a short README-level note).

### `make link-plugins` target

Add a new Makefile target `link-plugins` that detects the platform via `uname` and symlinks the installed Surge XT VST3 into `plugins/Surge XT.vst3`. This supersedes the current convention where Docker and Tart each create the symlink inline:

- Docker: `ln -s "/usr/lib/vst3/Surge XT.vst3" plugins/...`
- Tart: `ln -sfn '/Library/Audio/Plug-Ins/VST3/Surge XT.vst3' ~/synth-setter/'plugins/Surge XT.vst3'` (see `tart/macos.pkr.hcl:142`)

The target should:

- Detect Linux (canonical path: `/usr/lib/vst3/Surge XT.vst3`) and macOS (`/Library/Audio/Plug-Ins/VST3/Surge XT.vst3`, with `~/Library/...` as fallback).
- Create (or replace) the symlink at `plugins/Surge XT.vst3`.
- Print a clear error listing searched paths if Surge XT isn't found, so the user knows what to install.

The implementation should follow Option A from #487 (the child task tracking the cross-platform plugin path story). The PR that implements this target closes both this issue and #487.

### Getting-started alignment

- Update `docs/getting-started.md` §2 to match the canonical flow above.
- Remove redundant / contradictory content that duplicates the README.
- Cross-link to the Tart alternative (§2h) and the docker reference.
- Cross-link to the new env var docs section (sibling issue under #563).

## Out of scope

- W&B default-off behavior — tracked in the env var code overhaul sibling (#598).
- Env var docs rewrite (getting-started §4c) — tracked in the env var docs overhaul sibling.
- Tart Packer template update to call `make link-plugins` — tracked in the Tart Packer delegate sibling (#599); depends on this issue landing.

## Acceptance criteria

- [ ] A fresh Ubuntu 24.04 machine with Surge XT installed from the .deb can run the README install flow end-to-end and pass `make test`.
- [ ] A fresh macOS (Apple Silicon) machine with Surge XT installed from the cask can do the same.
- [ ] `make link-plugins` exists, works on Linux and macOS, and is referenced in the install walkthrough.
- [ ] README install section does not duplicate `docs/getting-started.md` §2 — each fact lives in exactly one place.
- [ ] README Codespaces / Docker / devcontainer-as-root content sits below the primary install + quick-start flow.
- [ ] `grep -rn "ln -s.*Surge XT.vst3" docker/ tart/ docs/` still finds references in Docker/Tart for now (those are updated by #599 and follow-up Docker PRs, not this issue).

Part of #264, Refs #563, supersedes the install flow half of #487 (which will be closed by the same PR).

### PR DESCRIPTION
**Pull request by ktinubu**
## Summary
- Rewrite README install section around a 5-step flow; `make install` is now the canonical end-to-end setup.
- `make install` is end-to-end: installs uv (if missing), creates `.venv/` with a uv-managed Python 3.10 interpreter (so users don't need Python pre-installed), installs `requirements.txt` + the project in editable mode, and registers pre-commit (skipped when `core.hooksPath` is set so it plays nicely with the dev container).
- Add `make link-plugins`: detects Linux/macOS and symlinks the installed Surge XT VST3 into `plugins/`, with explicit destination-safety handling (replaces symlinks/files, errors on real directory).
- Deduplicate README vs `docs/getting-started.md` §2 — README has a terse flow, getting-started has the walkthrough.
- Move Codespaces & Docker content below the primary install + quick-start flow in README.
- Add env-var export section and devcontainer-as-root note (pointing at the real `.devcontainer/{cpu,gpu}/devcontainer.json` paths).
- Streamline getting-started §2 to the uv-first canonical flow. Move the pip / conda / plain-venv walkthrough to a new **Appendix A: Manual environment setup** at the end of the doc.
- Drop `uv.lock`: we're committing to `uv pip` rather than `uv sync`. `uv sync` has known edge cases around torch indexes, transitive-dep resolution, and CPU/CUDA wheel selection that make it unsuitable here today.

## Test plan
- [x] Fresh Linux (Ubuntu 22.04 devcontainer) with Surge XT pre-installed: `make install` creates `.venv/` at Python 3.10.12 with prompt `synth-setter`, installs deps, `make test` → 155 passed / 2 skipped.
- [x] `make install` reruns idempotently (reuses `.venv/` if already 3.10).
- [x] `make install` errors when `.venv/` exists at a different Python version (tested with a 3.11 venv — clear error, suggests `rm -rf .venv`).
- [x] `make install` skips pre-commit install when `core.hooksPath` is set (dev container case).
- [x] `make link-plugins` creates symlink on Linux (`/usr/lib/vst3/Surge XT.vst3`).
- [x] `make link-plugins` destination safety: replaces existing symlink, replaces regular file, errors when destination is a real directory.
- [x] `make link-plugins` prints searched paths + install URL when Surge XT is not found.
- [x] `make help` lists the new `link-plugins` target.
- [x] No content duplication between README and getting-started §2 (README is terse, getting-started has the full walkthrough + appendix).
- [x] README ↔ getting-started anchors all resolve (`#appendix-a-manual-environment-setup`, `#2g-alternative-github-codespaces`, `#2h-alternative-local-dev-container`).
- [x] `grep -rn "ln -s.*Surge XT.vst3" docker/` still finds Dockerfile symlink references (ubuntu22_04/Dockerfile:173, :317).
- [ ] Fresh macOS (Apple Silicon) without Python pre-installed: `make install` → `make test` passes end-to-end (needs manual verification).
- [ ] Fresh Ubuntu 24.04 without uv pre-installed: `curl | sh` install branch of `make install` fires and succeeds (needs manual verification).

Closes #601
Closes #487

### CODE REVIEW

**Comment by ktinubu** [comment]
## PR Verification Checklist

- [x] **`make help` lists `link-plugins`:**
  ```
  link-plugins                   Symlink Surge XT VST3 into plugins/
  ```
- [x] **`make link-plugins` works on macOS:** `Linked plugins/Surge XT.vst3 -> /Library/Audio/Plug-Ins/VST3/Surge XT.vst3`
- [x] **Symlink correct:** `plugins/Surge XT.vst3 -> /Library/Audio/Plug-Ins/VST3/Surge XT.vst3`
- [x] **Docker symlink references preserved:**
  ```
  docker/ubuntu22_04/Dockerfile:173: ln -sfn "/usr/lib/${cmake_arch_name}-linux-gnu/vst3/Surge XT.vst3" ...
  docker/ubuntu22_04/Dockerfile:317: ln -s "/usr/lib/vst3/Surge XT.vst3" ...
  ```
- [x] **Taxonomy check:** passed (issue type, domain label, milestone present)
- [x] **`make format`:** all hooks pass
- [ ] **CI `check-pr-metadata`:** pending
- [ ] **CI `code-quality`:** pending
- [ ] **CI `Analyze (actions)`:** pending
- [ ] **CI `Analyze (python)`:** pending
- [ ] **Linux smoke test** (Ubuntu 24.04): not tested locally — needs CI or manual verification

File: README.md [hunk 1]
```diff
@@ -56,37 +56,33 @@ under the GPL-3.0 license.
...
 ```
 
-### Editable install
-
-```bash
-make install
-```
+> **pip / conda users:** If you manage your own Python environment, you can
+> skip steps 2-3 and run `make install` (or `pip install -r requirements.txt
+> -e .`) directly. See [docs/getting-started.md
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply README.md]: Fixed in 9977d84 — kept each inline code span on a single line within the blockquote.

File: docs/getting-started.md [hunk 2]
```diff
@@ -35,31 +49,56 @@ source .venv/bin/activate
...
-uv pip install -r requirements.txt -e .
+make install
 ```
 
-Or with plain pip:
+Or with plain pip (no editable install):
 
 ```bash
-pip install -r requirements.txt -e .
+pip install -r requirements.txt
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply docs/getting-started.md]: Fixed in 9977d84 — the plain-pip path now also runs `pip install -e .` so the project itself is installed (matching `make install`).

File: Makefile [hunk 3]
```diff
@@ -35,6 +35,34 @@ install: ## Install project in editable mode with dev deps
...
+		else \
+			echo "  /Library/Audio/Plug-Ins/VST3/$$PLUGIN_NAME"; \
+			echo "  $$HOME/Library/Audio/Plug-Ins/VST3/$$PLUGIN_NAME"; \
+		fi; \
+		echo "Install Surge XT from https://surge-synthesizer.github.io/"; \
+		exit 1; \
+	fi; \
+	mkdir -p plugins; \
+	ln -sfn "$$FOUND" "plugins/$$PLUGIN_NAME"; \
+	echo "Linked plugins/$$PLUGIN_NAME -> $$FOUND"
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply Makefile]: Fixed in 9977d84 — explicit destination handling: remove if it's a symlink or regular file, error out if it's a real directory, then create the symlink.

File: README.md [hunk 4]
```diff
@@ -184,6 +180,21 @@ Further reading (mostly for contributors and maintainers):
...
+GitHub Codespaces and local dev containers provide a pre-built environment with
+Surge XT, rclone, and all Python dependencies already installed. See
+[docs/getting-started.md &sect;2g](docs/getting-started.md#2g-alternative-github-codespaces)
+(Codespaces) and
+[&sect;2h](docs/getting-started.md#2h-alternative-local-dev-container)
+(local dev container) for setup instructions.
+
+> **Devcontainer as root:** The default dev container runs as a non-root user.
+> If your workflow requires root (e.g., installing system packages), set
+> `"remoteUser": "root"` in `.devcontainer/devcontainer.json`. See the
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply README.md]: Fixed in 9977d84 — points to the actual configs at `.devcontainer/cpu/devcontainer.json` and `.devcontainer/gpu/devcontainer.json`.

File: docs/getting-started.md [hunk 5]
```diff
@@ -9,12 +9,15 @@ ______________________________________________________________________
 ## 1. Prerequisites
 
 - **Linux or macOS** — Windows is not supported (see the project README).
-- **Python 3.10+** (check with `python --version`)
-- **Git**
-- **make** (ships with macOS/Linux)
+- **Git**, **curl**, **make** (all ship with macOS/Linux)
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply docs/getting-started.md]: Fixed in 3289289 — §1 now says make/curl/git are "standard on most macOS and Linux developer machines, but not guaranteed on minimal/server images" with a pointer to installing via the package manager.

File: Makefile [hunk 6]
```diff
@@ -31,9 +31,62 @@ test-full: ## Run all tests
...
+		[ "$$PY_VER" = "3.10" ] || { echo "ERROR: existing .venv has Python $$PY_VER, need 3.10 (rm -rf .venv to recreate)"; exit 1; }; \
+	else \
+		"$$UV" venv --python 3.10 --prompt synth-setter .venv; \
+	fi; \
+	"$$UV" pip install --python .venv/bin/python -r requirements.txt -e .; \
+	if [ -n "$$(git config --get core.hooksPath 2>/dev/null)" ]; then \
+		echo "Skipping pre-commit install (core.hooksPath is set; run '.venv/bin/pre-commit install' manually if you want to override)."; \
+	else \
+		.venv/bin/pre-commit install; \
+	fi
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply Makefile]: Fixed in 3289289 — the install recipe now prepends `set -e`, so any failure in uv, pip, or pre-commit propagates and aborts the target.

File: Makefile [hunk 7]
```diff
@@ -31,9 +31,117 @@ test-full: ## Run all tests
...
+		echo "  actual:   $$ACTUAL_MD5"; \
+		echo "Remove the cached file and retry: rm '$$ARCHIVE'"; \
+		exit 1; \
+	fi; \
+	echo "md5 OK. Extracting Surge XT.vst3 into plugins/..."; \
+	case "$$OS" in \
+		Linux) tar -xzf "$$ARCHIVE" -C plugins/ "./Surge XT.vst3" ;; \
+		Darwin) unzip -q "$$ARCHIVE" "Surge XT.vst3/*" -d plugins/ ;; \
+	esac; \
+	echo "Installed $$DEST"
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply Makefile:112]: Fixed in 3289289 — `install-surge-xt` now prepends `set -e`, so tar/unzip failure aborts the recipe before the "Installed …" echo can mask it.

File: Makefile [hunk 8]
```diff
@@ -31,9 +31,117 @@ test-full: ## Run all tests
...
+	if [ ! -f "$$ARCHIVE" ]; then \
+		echo "Downloading $(SURGE_XT_RELEASE_URL)/$$ASSET"; \
+		curl -fSL -o "$$ARCHIVE" "$(SURGE_XT_RELEASE_URL)/$$ASSET"; \
+	else \
+		echo "Using cached $$ARCHIVE"; \
+	fi; \
+	if command -v md5sum >/dev/null 2>&1; then \
+		ACTUAL_MD5=$$(md5sum "$$ARCHIVE" | awk '{print $$1}'); \
+	else \
+		ACTUAL_MD5=$$(md5 -q "$$ARCHIVE"); \
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply Makefile]: Fixed in 3289289 — md5 detection now has an explicit elif/else: `md5sum` → `md5` → otherwise error with "neither md5sum nor md5 is available — cannot verify checksum" and exit 1.

File: docs/getting-started.md [hunk 9]
```diff
@@ -26,40 +29,69 @@ git clone https://github.com/tinaudio/synth-setter.git
...
+guide assume the venv is active.
+
+### 2d. Symlink the Surge XT VST3
+
+If you have [Surge XT](https://surge-synthesizer.github.io/) installed, symlink
+it into `plugins/` so the test suite and data pipeline can find it:
 
 ```bash
-pip install -r requirements.txt -e .
+make link-plugins
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply docs/getting-started.md]: Fixed in 3289289 — §2d now includes a "heads-up" call-out naming `tests/data/vst/test_preset_params.py` and `tests/docker/test_smoke.py` as hardcoding `/usr/lib/vst3/Surge XT.vst3`, with a pointer to the follow-up issue. Tracked separately in #631 (Bug under Epic #149 Phase 3), which will default the tests to `plugins/Surge XT.vst3` with an env-var override for CI.

File: Makefile [hunk 10]
```diff
@@ -31,9 +31,117 @@ test-full: ## Run all tests
...
+	DEST="plugins/$$PLUGIN_NAME"; \
+	if [ -L "$$DEST" ]; then \
+		rm -f "$$DEST"; \
+	elif [ -d "$$DEST" ]; then \
+		echo "ERROR: $$DEST exists as a directory; remove it before linking."; \
+		exit 1; \
+	elif [ -e "$$DEST" ]; then \
+		rm -f "$$DEST"; \
+	fi; \
+	ln -s "$$FOUND" "$$DEST"; \
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply Makefile]: Moot in 3289289 — same as above, `link-plugins` removed.

File: docs/getting-started.md [hunk 11]
```diff
@@ -26,40 +29,69 @@ git clone https://github.com/tinaudio/synth-setter.git
...
+### 2b. Install
+
+`make install` is the canonical end-to-end install. It:
+
+1. Installs [uv](https://docs.astral.sh/uv/) if it is not already on your PATH.
+2. Creates `.venv/` using a managed Python 3.10 interpreter (downloaded by uv
+   if you do not have one locally). The venv prompt label is `synth-setter`.
+3. Installs everything in `requirements.txt` plus the project itself in
+   editable mode (`pip install -e .`).
+4. Registers the pre-commit hooks.
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply docs/getting-started.md]: Fixed in 3289289 — §2b now spells out that pre-commit install is skipped when `core.hooksPath` is set, with instructions for manual override.

File: Makefile [hunk 12]
```diff
@@ -31,9 +31,62 @@ test-full: ## Run all tests
...
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply Makefile]: Moot in 3289289 — the `link-plugins` target was removed entirely. The replacement is a one-line `ln -s` in the docs, which users run directly.

File: Makefile [hunk 13]
```diff
@@ -31,9 +31,117 @@ test-full: ## Run all tests
...
+link-plugins: ## Symlink an existing system-wide Surge XT VST3 into plugins/
+	@PLUGIN_NAME="Surge XT.vst3"; \
+	OS=$$(uname -s); \
+	FOUND=""; \
+	if [ "$$OS" = "Linux" ]; then \
+		if [ -e "/usr/lib/vst3/$$PLUGIN_NAME" ]; then FOUND="/usr/lib/vst3/$$PLUGIN_NAME"; fi; \
+	elif [ "$$OS" = "Darwin" ]; then \
+		for p in "/Library/Audio/Plug-Ins/VST3/$$PLUGIN_NAME" "$$HOME/Library/Audio/Plug-Ins/VST3/$$PLUGIN_NAME"; do \
+			if [ -e "$$p" ]; then FOUND="$$p"; break; fi; \
+		done; \
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply Makefile]: Moot in 3289289 — `link-plugins` was removed. Users with Surge XT installed anywhere run `ln -s "/your/path/Surge XT.vst3" "plugins/Surge XT.vst3"` directly.

File: README.md [hunk 14]
```diff
@@ -48,45 +48,37 @@ under the GPL-3.0 license.
...
-```
-
-### Using uv (recommended)
 
-```bash
-uv pip install -r requirements.txt
-```
-
-### Using pip
+# 2. Install uv, create .venv (Python 3.10), install deps, set up pre-commit
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply README.md]: Fixed in 3289289 — the step-4 code comment now mentions the pre-commit skip when `core.hooksPath` is set (dev container case).
